custom/icds_reports/utils/__init__.py

```diff
@@ -1774,3 +1777,82 @@ def create_group_by(aggregation_level):
 
 def column_value_as_per_agg_level(aggregation_level, agg_level_threshold, true_value, false_value):
     return true_value if aggregation_level > agg_level_threshold else false_value
+
+
+@attr.s
+class AggLevelInfo(object):
+    agg_level = attr.ib()
+    col_name = attr.ib()
+
+
+def _construct_repacement_map_from_awc_location(loc_level, replacement_location_ids):
+    levels = {
+        'state': AggLevelInfo(1, 'state'),
+        'district': AggLevelInfo(2, 'district'),
+        'block': AggLevelInfo(3, 'block'),
+        'supervisor': AggLevelInfo(4, 'supervisor'),
+        'awc': AggLevelInfo(5, 'doc')
+    }
+    level_info = levels[loc_level]
+    filters = {
+        f'{level_info.col_name}_id__in': replacement_location_ids,
+        'aggregation_level': level_info.agg_level
+    }
+    columns = [
+        f'{level_info.col_name}_id',
+        'state_name',
+        'district_name',
+        'block_name',
+        'supervisor_name',
+        'awc_name'
+    ]
+    replacements = AwcLocation.objects.filter(**filters).values(*columns)
+
+    def _full_hierarchy_name(loc):
+        loc_names = [loc[f'{level}_name'] for level in levels.keys() if loc[f'{level}_name']]
+        return ' > '.join(loc_names)
+
+    replacement_names = {loc[f'{level_info.col_name}_id']: _full_hierarchy_name(loc) for loc in replacements}
```

Comment:
This is just a simple map of `{location_id: location_name}`. What makes it different than just putting the name is that it includes the whole hierarchy of that location, so that users can find it in the location filter. It looks like `'state > district > block'`. I can add that in a comment, if you think that would help since its obviously not clear thats what it does, but an object seems like overkill

Reply:
sorry, that seems fine, I got confused by [this line](https://github.com/dimagi/commcare-hq/pull/27166/files#diff-c87f5ffa4bc46d5bec44951bdde6be41R1847) and thought it had some additional structure, but see now that's coming from `location.metadata`